### PR TITLE
[Shipping] Rework the Shipping component to use ServiceRegistry

### PR DESCRIPTION
**Pull request by tuka217**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Fixed tickets | -
| License       | MIT
| Doc PR        | #367

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
@tuka217 Please have a look at the failing build. :)

File: src/Sylius/Bundle/CoreBundle/CHANGELOG.md [hunk 1]
```diff
@@ -7,3 +7,7 @@ CHANGELOG
 * Twig extension was renamed from `SyliusRestrictedZoneExtension` into `RestrictedZoneExtension`,
   also the service name was changed from `sylius.twig.restricted_zone_extension`
   to `sylius.twig.extension.restricted_zone`.
+
+### v0.16.0
+
+* Change calculatorRegistry type to `ServiceRegistryInterface` in `ShippingMethodTypeSpec`
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/CoreBundle/CHANGELOG.md]: I'd say ``Replaced custom RuleCheckerRegistry and CalculatorRegistry with ServiceRegistry component.``

**Comment by pjedrzejewski** [comment]
Perfect, thank you Ania! I like getting rid of code. :)

File: src/Sylius/Bundle/CoreBundle/CHANGELOG.md [hunk 2]
```diff
@@ -7,3 +7,7 @@ CHANGELOG
 * Twig extension was renamed from `SyliusRestrictedZoneExtension` into `RestrictedZoneExtension`,
   also the service name was changed from `sylius.twig.restricted_zone_extension`
   to `sylius.twig.extension.restricted_zone`.
+
+### v0.16.0
+
+*Replaced custom  CalculatorRegistry with ServiceRegistry component in ShippingMethodTypeSpec
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/CoreBundle/CHANGELOG.md]: Missing space after `*`

**Pull request merged**